### PR TITLE
fix(tooltip): getTooltipContent should avoid undefined

### PR DESCRIPTION
**Pull request by HHongSeungWoo**
## Description

This is a follow-up fix for #3932
addressing an error occurring when data is undefined

## Related Issue

https://github.com/recharts/recharts/pull/3932#issuecomment-1809139750

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/chart/generateCategoricalChart.tsx [hunk 1]
```diff
@@ -233,7 +233,7 @@ const getTooltipContent = (
      * Defaulting to chartData below to fix an edge case where the tooltip does not include data from all charts
      * when a separate dataset is passed to chart prop data and specified on Line/Area/etc prop data
      */
-    const data = (child.props.data ?? chartData).slice(state.dataStartIndex, state.dataEndIndex + 1);
+    const data = (child.props.data ?? chartData)?.slice(state.dataStartIndex, state.dataEndIndex + 1);
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:236]: Could you please add a test to prevent the bug from coming up again?

Comment (HHongSeungWoo) [reply src/chart/generateCategoricalChart.tsx:236]: Testing this function is not easy, i wating for a reproducible example.
https://github.com/recharts/recharts/pull/3932#issuecomment-1809236675

Comment (nikolasrieble) [reply src/chart/generateCategoricalChart.tsx:236]: Could we extract this function as an independent helper function? Possibly a follow up task.

**Review by nikolasrieble — APPROVED** [review]
The fix is trivial and solves the bug. Lets land it.

**Review by ckifer — APPROVED** [review]
